addons/event/views/event_registration_views.xml

```diff
@@ -246,18 +249,41 @@
                             invisible="context.get('registration_view_hide_group_by_create_date_week')"/>
                     <filter string="Registration Date" name="group_by_create_date_day" domain="[]" context="{'group_by': 'create_date:day'}"
                             invisible="not context.get('registration_view_hide_group_by_create_date_week')"/>
+                    <filter string="Campaign" name="group_by_utm_campaign_id" domain="[]"
+                            context="{'group_by': 'utm_campaign_id'}"/>
+                    <filter string="Medium" name="group_by_utm_medium_id" domain="[]"
+                            context="{'group_by': 'utm_medium_id'}"/>
+                    <filter string="Source" name="group_by_utm_source_id" domain="[]"
+                            context="{'group_by': 'utm_source_id'}"/>
                </group>
             </search>
         </field>
     </record>
 
+    <!-- Search view specific for an event so remove event related fields and filters -->
```

Comment:
Maybe rephrase a bit?

Sugg:

"Search view typically used when coming from a specific event, meaning we don't need event-related fields"

Reply:
Yes better, thanks